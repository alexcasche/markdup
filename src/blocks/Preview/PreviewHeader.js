import styled from 'styled-components'
import { tint } from 'polished'
import { applyStyleModifiers } from 'styled-components-modifiers'

const MODIFIERS = {
  expanded: () => `
    position: fixed;
    top: 0;
    left: 0;
    right: 0;
    padding: 0 10vw;
    z-index: 1;
  `,
}

const PreviewHeader = styled.header`
  display: flex;
  align-items: center;
  width: 100%;
  padding: 0 15px;
  background-color: #f7f7f7;
  border-top: 1px solid ${props => tint(0.5, props.theme.colors.grey)};
  border-bottom: 1px solid ${props => tint(0.5, props.theme.colors.grey)};
  overflow: visible;
  min-height: 46px;
  ${applyStyleModifiers(MODIFIERS)};
`

export default PreviewHeader
