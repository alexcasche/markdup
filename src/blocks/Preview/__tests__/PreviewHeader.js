import PreviewHeader from '../PreviewHeader'

describe('PreviewHeader Element', () => {
  it('renders correctly', () => {
    const tree = render(<PreviewHeader>Preview</PreviewHeader>).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('adds expanded modifier', () => {
    const tree = render(<PreviewHeader modifiers={['expanded']} />).toJSON()
    expect(tree).toMatchSnapshot()
    expect(tree).toHaveStyleRule('padding', '0 10vw')
  })
})
