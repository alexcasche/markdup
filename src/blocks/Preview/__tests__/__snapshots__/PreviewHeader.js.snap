// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PreviewHeader Element adds expanded modifier 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  padding: 0 15px;
  background-color: #f7f7f7;
  border-top: 1px solid #d5d6d9;
  border-bottom: 1px solid #d5d6d9;
  overflow: visible;
  min-height: 46px;
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  padding: 0 10vw;
  z-index: 1;
}

<header
  className="c0"
/>
`;

exports[`PreviewHeader Element renders correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  padding: 0 15px;
  background-color: #f7f7f7;
  border-top: 1px solid #d5d6d9;
  border-bottom: 1px solid #d5d6d9;
  overflow: visible;
  min-height: 46px;
}

<header
  className="c0"
>
  Preview
</header>
`;
